mne_bids/io.py

```diff
@@ -51,6 +59,24 @@ def _read_raw(raw_fname, electrode=None, hsp=None, hpi=None, config=None,
     elif ext == '.ds':
         raw = io.read_raw_ctf(raw_fname)
 
+    # EEG File Types
+    # --------------
+    # BrainVision format by Brain Products, links to  a .eeg and a .vmrk file
+    elif ext == '.vhdr':
+        raw = io.read_raw_brainvision(raw_fname)
+
+    # EDF (european data format) or BDF (biosemi) format
+    elif ext == '.edf' or ext == '.bdf':
+        raw = io.read_raw_edf(raw_fname, preload=True)
```

Comment:
why is using preload=True necessary?

Reply:
Because MNE-Python wants it so for the EDF reader, else you get:

`RuntimeError: EDF+ Annotations (TAL) channel needs to be parsed completely on loading. You must set preload parameter to True.`

Reply:
ok fair enough.

I get we need to fix this in mne-python. We should not need to preload to get the raw.annotations available.

cc @massich